src/config.rs

```diff
@@ -223,6 +224,112 @@ impl Config {
+        fn segregate_key_value(
+            header: &str,
+            doc_id: &str,
+            line_number: usize,
+        ) -> ftd::p1::Result<(String, String)> {
+            if !header.contains(':') {
+                return Err(ftd::p1::Error::ParseError {
+                    message: format!(": is missing in: {}", header),
+                    doc_id: doc_id.to_string(),
+                    line_number,
+                });
+            }
+
+            let mut parts = header.splitn(2, ':');
+            match (parts.next(), parts.next()) {
+                (Some(name), Some(value)) => {
+                    // some header and some non-empty value
+                    Ok((name.trim().to_string(), value.trim().to_string()))
+                }
+                (Some(name), None) => Err(ftd::p1::Error::ParseError {
+                    message: format!("Unknown header value for header \'{}\'", name),
+                    doc_id: doc_id.to_string(),
+                    line_number,
+                }),
+                _ => Err(ftd::p1::Error::ParseError {
+                    message: format!("Unknown header found \'{}\'", header),
+                    doc_id: doc_id.to_string(),
+                    line_number,
+                }),
+            }
+        }
+
+        /// converts the given string to document_id
+        /// and returns it
+        fn convert_to_document_id(doc_id: &str) -> String {
+            // Todo use lazystatic! for this regex
+            let ext: regex::Regex = regex::Regex::new(r".[a-z\d]+[/]?$").unwrap();
+            let doc_id = ext.replace_all(doc_id, "");
+
+            // Discard document suffix if there
+            // Also discard trailing index
+            let document_id = doc_id
+                .split_once("/-/")
+                .map(|x| x.0)
+                .unwrap_or_else(|| doc_id.as_ref())
+                .trim_end_matches("index")
+                .trim_matches('/');
+
+            // In case if doc_id = index.ftd
+            if document_id.is_empty() {
+                return "/".to_string();
+            }
+
+            // Attach /{doc_id}/ before returning
+            return format!("/{}/", document_id);
+        }
+
+        /// updates the config.terms map
+        ///
+        /// mapping from [term -> document_id]
+        fn update_terms(
+            terms_map: &mut std::collections::HashMap<String, String>,
+            term_string: &str,
+            doc_id: &str,
+            line_number: usize,
+        ) -> ftd::p1::Result<()> {
+            let (_header, term) = segregate_key_value(term_string, doc_id, line_number)?;
+
+            // let slugified_term = slug::slugify(term);
+            // dbg!(doc_id);
+            let document_id = convert_to_document_id(doc_id);
+            // dbg!(&document_id);
+            // println!("updated map!!");
+            // dbg!(&slugified_term, &document_id);
+
+            terms_map.insert(term, document_id);
```

Comment:
- Check if `term` already exists, if yes then throw the appropriate error.
- Keep the full URL of the term i.e. (<document-id>/#<slugify-term>). Do the related changes in ftd pr too.